src/Category/Service/CategoryProductsSortingOptionsProvider.php

```diff
@@ -39,13 +39,17 @@ public function getAllSortingOptions(): array
 
             $sortableFields = $this->sourceFieldRepository->getSortableFields('product');
             foreach ($sortableFields as $sourceField) {
-                /** @var ProductSortOrderProviderInterface $sortOrderProvider */
-                foreach ($this->sortOrderProviders as $sortOrderProvider) {
-                    if ($sortOrderProvider->supports($sourceField)) {
-                        $sortOptions[] = [
-                            'code' => $sortOrderProvider->getSortOrderField($sourceField),
-                            'label' => $sortOrderProvider->getSimplifiedLabel($sourceField),
-                        ];
+                // Id source field need to be sortable to be used as default sort option,
+                // but we don't want to have it in the list
+                if ('id' !== $sourceField->getCode()) {
```

Comment:
We should maybe add an extra parameter  on the function getSortableFields('product', ['excluded_source_field_code']) ,  to avoid to get useless data ?

or replace by code above , it's maybe more readable ?
```
if ('id' === $sourceField->getCode()) {
    // Id source field need to be sortable to be used as default sort option,
    // but we don't want to have it in the list
    continue;
}
```